RoddyCore/src/de/dkfz/roddy/knowledge/files/BaseFile.groovy

```diff
@@ -549,16 +553,17 @@ abstract class BaseFile<FS extends FileStageSettings> extends FileObject {
         parentFiles.add(baseFile)
     }
 
+    @Deprecated
     void setParentFiles(List<BaseFile> parentfiles) {
-        setParentFiles(parentfiles, false)
+        setParentFiles(parentfiles, false, DEFAULT)
     }
 
-    void setParentFiles(List<BaseFile> parentfiles, boolean resetFilename) {
+    void setParentFiles(List<BaseFile> parentfiles, boolean resetFilename, String selectionTag) {
```

Comment:
I am wondering whether handing over `selectionTag` from method to method should be classified as bad smell and an indication of a design issue ...
Should it maybe be a field of BaseFile?

Reply:
As far as I remember, in some cases, the tag is overriden by the workflow developer, that is why you can pass it.